Allow routes to define some payload config values

afharo opened the pull request:
## Summary

When defining the routes, the plugins can now specify 4 additional optional parameters:
- `accepts`: https://hapi.dev/api/?v=18.4.0#-routeoptionspayloadallow
- `parse`: https://hapi.dev/api/?v=18.4.0#-routeoptionspayloadparse
- `output`: https://hapi.dev/api/?v=18.4.0#-routeoptionspayloadoutput
- `maxBytes`: https://hapi.dev/api/?v=18.4.0#-routeoptionspayloadmaxbytes

Fixes #48470 and #41960

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- ~~[ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
- ~~[ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- [X] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [X] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- ~~[ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [X] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [X] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

afharo committed:
Allow routes to define some payload config values

elasticmachine commented:
Pinging @elastic/kibana-platform (Team:Platform)

kobelb reviewed (approved):
Security changes 👍

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/10080/)
* Commit: 56185aa46c64bd6b59d7f5e19d459156672f834d

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/10099/)
* Commit: 7cbea0c7f45b760415cb79e0e27a31f4ba92c24e

pgayvallet commented on src/core/server/http/router/request.ts:
```diff
@@ -33,10 +33,12 @@ const requestSymbol = Symbol('request');
  * @public
  * */
-export interface KibanaRequestRoute {
+export interface KibanaRequestRoute<Method extends RouteMethod | 'patch' | 'options'> {
```
Question @restrry: as they seems supported, why aren't `patch` and `options` actually in our `RouteMethod` type ?

restrry replied:
Probably because there are no use cases in the legacy platform. The initial implementation was done by @azasypkin

pgayvallet replied:
Should we add them to `RouteMethod` then, or do you see any reason not to do so?

restrry replied:
I'd avoid extending API surface until it's necessary. YMMV

pgayvallet commented on src/core/server/http/router/route.ts:
```diff
@@ -24,11 +24,23 @@ import { ObjectType } from '@kbn/config-schema';
+export type RouteContentType =
+  | 'application/json'
+  | 'application/*+json'
+  | 'application/octec-stream'
```
Typo: `application/octec-stream` -> `application/octet-stream`

afharo replied:
Good catch! Changed

pgayvallet commented on src/core/server/http/router/route.ts:
```diff
@@ -42,13 +54,64 @@ export interface RouteConfigOptions {
+   * Default value: The one set in the kibana.yml config file under the parameter `server.maxPayloadBytes`.
+   * Limits the size of incoming payloads to the specified byte count. Allowing very large payloads may cause the server to run out of memory.
+   */
+  maxBytes?: Method extends 'get' ? never : number;
```
NIT: I would have put the default value explanation after the actual description of the fields.

afharo replied:
I agree! The pre-existing params provide the default value at the end. Updated.

pgayvallet reviewed (commented):
Congrats on your first PR 😄

afharo commented on src/core/server/http/http_server.ts:
```diff
@@ -127,21 +127,32 @@ export class HttpServer {
             validate,
+            payload: [allow, maxBytes, output, parse].some(v => typeof v !== 'undefined')
+              ? { allow, maxBytes, output, parse }
+              : undefined,
```
I tried that before coming up with this solution, unfortunately, HAPI does not allow `payload` options for HEAD or GET requests (https://github.com/hapijs/hapi/blob/803f94331b13f7d1747f7f99b072d57574fd65fe/lib/route.js#L61), so setting the object, despite everything inside it is `undefined`, HAPI will reject it.

I considered filtering by method (like in https://github.com/elastic/kibana/blob/7cbea0c7f45b760415cb79e0e27a31f4ba92c24e/src/core/server/http/http_server.ts#L131) but I felt like it might cause confusion when, for whatever reason, the user decides to set any of these parameters when defining a GET route/request, leading to unexpected behaviour.

I thought it would be more convenient to go down the route of documenting it as "you can't do that" via TS (as you may have already seen in this MR) 🙃

pgayvallet replied:
Ok. That answers my question about `GET` payloads.

afharo replied:
That was intended. I thought it was better to let them get that error vs. simply removing it for them, leading to unexpected behaviour (if the dev is explicitly setting that config, maybe he means it for something?).

Do you think we should do the validation ourselves and throw the error even earlier?

restrry replied:
we can add our custom validation. but as I said it's not big deal.

afharo committed:
Documentation typo

afharo commented:
@elasticmachine merge upstream

azasypkin commented on src/core/server/http/router/route.ts:
```diff
@@ -42,13 +54,68 @@ export interface RouteConfigOptions {
+   *
+   * Default value: 'data'.
+   */
+  output?: Method extends 'get' ? never : 'data' | 'stream' | 'file';
```
__question:__ These questions are just for my own understanding, please feel free to ignore (don't want to block as I'm OOO anyway :slightly_smiling_face: ):

* `.. specified by the uploads settings` --- > Do we use `file` anywhere in Kibana already? If not, maybe it'd make sense to not expose this for now to keep exposed API surface manageable and less dependent on Hapi? True for all other options we're exposing here.
* I always felt that some of the Hapi names are really confusing, can we use our own more descriptive names since it's a brand new API for core anyway (e.g. `output` ---> `payloadFormat`, `bodyFormat` or something similar)?
* I feel a bit worried that `parse` isn't tied to `validate` in any way. I believe you had discussion around that already and I'm just missing context, but can't we set `parse` to `false` if `validate` is set to `false` for example or at least set better/safer defaults (e.g. `parse: false` and `output: 'stream'`).
* Do we validate at run-time that all these parameters corresponds to what we require via types or we solely rely on Hapi to validate those? If the latter we may be in troubles in the future when we upgrade Hapi and it introduces new options we don't want to support or just handle differently. 

As you see I'm just alluding to the fact that we may not want to copy Hapi APIs and instead model our own public API with better defaults that would rely only on certain parts of the Hapi APIs under the hood. /cc @restrry @kobelb

afharo replied:
Hi @azasypkin, thank you for your comments, I really appreciate them.

> `.. specified by the uploads settings` --- > Do we use `file` anywhere in Kibana already? If not, maybe it'd make sense to not expose this for now to keep exposed API surface manageable and less dependent on Hapi? True for all other options we're exposing here.

I agree. Since it is not currently used, maybe it's better to hide it from the docs (and use a validation step to reject it if provided despite types?)

> I always felt that some of the Hapi names are really confusing, can we use our own more descriptive names since it's a brand new API for core anyway (e.g. `output` ---> `payloadFormat`, `bodyFormat` or something similar)?

This is kind of my fault, the actual property from HAPI is `payload.output`. Maybe it provides more context. I brought these 4 properties up to the top-level options thinking this way we don't expose the raw `payload` option to the user (where they could do `{anyOtherNotExposedOption} as any` to workaround us). But I can group them down to `payload` and still write the code in a defensive way.

> I feel a bit worried that `parse` isn't tied to `validate` in any way. I believe you had discussion around that already and I'm just missing context, but can't we set `parse` to `false` if `validate` is set to `false` for example or at least set better/safer defaults (e.g. `parse: false` and `output: 'stream'`).

I like the idea of setting `parse: false` and `output: 'stream'` if `validate === false`. That should release some memory pressure. I wonder what we should do when `validate` exists but doesn't contain the `body` property?

> Do we validate at run-time that all these parameters corresponds to what we require via types or we solely rely on Hapi to validate those? If the latter we may be in troubles in the future when we upgrade Hapi and it introduces new options we don't want to support or just handle differently.

Hapi does some validation under the hood for "sensible" values, but you are right. We should be more defensive regarding our accepted values (over the wider options provided by Hapi). I'll see what I can do with this.

afharo replied:
Updated, taking into account this conversation:
1. Moved the payload-specific config inside a property called `body` to match the same naming convention we've got in `validate`
2. Added validation to avoid `body.output` to be specified as anything different to 'data' or 'stream'.
3. When `validate.body` does not exist, we assume the route does not care about the body content, so we default to `output: 'stream'` and `parse: false`

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/10335/)
* Commit: aceb9371ca28f87974afdef025b817b52d71ca46

afharo committed:
Move hapi `payload` config under `body` + additional validations

afharo committed:
Update API docs

afharo committed:
Amend explanation in API docs

afharo reviewed (commented):
Updated, taking into account [this conversation](https://github.com/elastic/kibana/pull/50783#discussion_r347314519):
1. Moved the payload-specific config inside a property called `body` to match the same naming convention we've got in `validate`
2. Added validation to avoid `body.output` to be specified as anything different to `'data'` or `'stream'`. I didn't find any of the other 3 values might produce any harm if extended by HAPI in the future (unless they go really mad 😅).
3. When `validate.body` does not exist, we assume the route does not care about the body content, so we default to `output: 'stream'` and `parse: false`

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/10428/)
* Commit: 0312e410b26cba2f97e672f86fcbd1fbba8b9984

restrry commented on src/core/server/http/router/request.ts:
```diff
@@ -33,10 +33,12 @@ const requestSymbol = Symbol('request');
+  method: Method;
+  options: Method extends 'get'
+    ? Required<Pick<RouteConfigOptions<Method>, 'authRequired' | 'tags'>> // GET requests cannot have payload options
+    : Required<RouteConfigOptions>; // Using any because 'patch' and 'options' are not valid RouteMethods but shouldn't affect much to typings
```
not only Hapi. https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/console/server/request.ts#L37-L39

afharo replied:
Hmm, should we create a task to fork HAPI/change the request lib in NP to support these queries?

restrry replied:
not sure we will start working on this any time soon, but probably it's good to document this limitation.

restrry commented on src/core/server/http/router/request.ts:
```diff
@@ -33,10 +33,12 @@ const requestSymbol = Symbol('request');
-  options: Required<RouteConfigOptions>;
+  method: Method;
+  options: Method extends 'get'
+    ? Required<Pick<RouteConfigOptions<Method>, 'authRequired' | 'tags'>> // GET requests cannot have payload options
```
requires changes when `RouteConfigOptions`  adds new options. shouldn't we Omit `body` instead?

afharo replied:
You are totally right! Changing it now.

restrry commented on src/core/server/http/router/route.ts:
```diff
@@ -24,11 +24,74 @@ import { ObjectType } from '@kbn/config-schema';
+   * * true - if the request 'Content-Type' matches the allowed mime types set by allow (for the whole payload as well as parts), the payload is converted into an object when possible. If the
+   * format is unknown, a Bad Request (400) error response is sent. Any known content encoding is decoded.
+   * * false - the raw payload is returned unmodified.
+   * * 'gunzip' - the raw payload is returned unmodified after any known content encoding is decoded.
```
I see only one usage. Can we switch to `true` instead? `'gunzip'` looks to generic and hapi specific

afharo replied:
Unfortunately, I don't think we can use `true` in [that one usage](https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/elasticsearch/lib/create_proxy.js#L31). It's related to the (multi-search endpoint)[https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html] in ElasticSearch where we are receiving NDJSON format in the form of
```
header\n
body\n
header\n
body\n
```
Example:
```
{"index" : "test"}
{"query" : {"match_all" : {}}, "from" : 0, "size" : 10}
{"index" : "test", "search_type" : "dfs_query_then_fetch"}
{"query" : {"match_all" : {}}}
{}
{"query" : {"match_all" : {}}}

{"query" : {"match_all" : {}}}
{"search_type" : "dfs_query_then_fetch"}
{"query" : {"match_all" : {}}}
```
I don't think HAPI knows how to parse that, and using  `false`, we may risk passing through to ElasticSearch a compressed payload that it may not understand.

restrry replied:
I see. Optional: Shouldn't we rename it to something like `try`. Not to expose specific implementation?

restrry commented on src/core/server/http/router/router.ts:
```diff
@@ -138,6 +139,43 @@ function routeSchemasFromRouteConfig<
+  const shouldValidateBody = validate && validate.body;
+
+  const { output } = options.body || {};
+  if (typeof output === 'string' && !['data', 'stream'].includes(output)) {
```
nit: easy to forget to update this logic. Shouldn't we extract in own type / enum? https://github.com/elastic/kibana/pull/50783/files#diff-3a561a752ff7c0106ea1ce21c8a9ea49R76

afharo replied:
Makes perfect sense! Let me change this

restrry commented on src/core/server/http/router/router.ts:
```diff
@@ -138,6 +139,43 @@ function routeSchemasFromRouteConfig<
+    throw new Error(
+      `[options.body.output: '${output}'] in route ${method.toUpperCase()} ${
+        routeConfig.path
+      } is not valid. Only 'data' or 'stream' are valid.`
```
super nit: probably worth referencing valid options from code instead.

mshustov reviewed (commented):
It would be good to add several integration tests for introduced options. It would help us if we update the underlying `hapi` version or migrate to another library.

afharo committed:
Add stream and buffer types to @kbn/config-schema

afharo committed:
Fixes based on PR feedback:
- Add 'patch' and 'options' to valid RouteMethod
- Add tests for all the new flags
- Allow `stream` and `buffer` schema in the body validations (findings from tests)

afharo committed:
API documentation update

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/10980/)
* Commit: 6199d84ba7660bce735a504ae6de7859fcc4b423

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/10999/)
* Commit: 07eae13cc3d7b1a7fdc090d150f2dfaf0ec5943f

afharo committed:
Fix type definitions

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/11063/)
* Commit: cadfcf941224b6f4c859d46a5a4e1ceefcc505fd

restrry commented on src/core/server/http/http_server.test.ts:
```diff
@@ -577,6 +578,173 @@ test('exposes route details of incoming request to a route handler', async () =>
+        try {
+          expect(req.body).toBeInstanceOf(Buffer);
+          expect(req.body.toString()).toBe(JSON.stringify({ test: 1 }));
+          return res.ok({ body: req.route });
```
we will have false negatives if change `route` format. shouldn't we pass only data to test?

restrry commented on docs/development/core/server/kibana-plugin-server.irouter.options.md:
````diff
@@ -0,0 +1,13 @@
+<b>Signature:</b>
+
+```typescript
+options: <P extends ObjectType, Q extends ObjectType, B extends ObjectType>(route: RouteConfig<P, Q, B, 'options'>, handler: RequestHandler<P, Q, B>) => void;
````
Is there are real use case when a plugin wants to declare a route handler for `options` request?

afharo replied:
Not that I know of, but I implemented it only for completion because of this discussion: https://github.com/elastic/kibana/pull/50783#discussion_r346995159

But I just saw your comment in there to avoid extending the API. Reverting those changes then :)

restrry commented on packages/kbn-config-schema/src/internals/index.ts:
```diff
@@ -89,6 +90,33 @@ export const internals = Joi.extend([
+    base: Joi.binary(),
+    coerce(value: any, state: State, options: ValidationOptions) {
+      // If value isn't defined, let Joi handle default value if it's defined.
+      if (value !== undefined && !(typeof value === 'object' && value instanceof Buffer)) {
```
[Buffer.isBuffer](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_isbuffer_obj)

mshustov reviewed (approved):
some nits, LGTM overall. :shipit: congrats with the fist PR in the NP 🍻

afharo committed:
Fix the NITs in the PR comments + better typing inheritance

afharo committed:
API docs update

afharo committed:
Fix APM-legacy wrapper's types

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/11542/)
* Commit: fa910a893d765950dbde3c7226c74ba86ab6839c

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/11562/)
* Commit: 7d4afae6a283634d65f6987f26ad340d946822ca

azasypkin commented on packages/kbn-config-schema/src/index.ts:
```diff
@@ -65,6 +68,14 @@ function boolean(options?: TypeOptions<boolean>): Type<boolean> {
   return new BooleanType(options);
 }
 
+function buffer(options?: TypeOptions<Buffer>): Type<Buffer> {
```
__note:__ please don't forget to document these new methods once https://github.com/elastic/kibana/pull/50307/ is merged.

afharo commented:
@elastic/apm-ui can you have a look at these changes when you have a moment? Your review is blocking this merge to happen

dgieselaar reviewed (approved):
I'll approve, with the caveat that I'm stumped as to why it doesn't work without the type assertion 🤷🏻‍♂️.

restrry commented on docs/development/core/server/kibana-plugin-server.basepath.get.md:
````diff
@@ -9,5 +9,5 @@ returns `basePath` value, specific for an incoming request.
 
 ```typescript
-get: (request: KibanaRequest<unknown, unknown, unknown> | LegacyRequest) => string;
+get: (request: KibanaRequest<import("hapi").Util.HTTP_METHODS_PARTIAL_LOWERCASE, unknown, unknown, unknown> | LegacyRequest) => string;
````
Hapi leaking in our docs is not good. why not declare union type manually? `'get' | 'post' | ...`

afharo replied:
Fixed by simply adding a better default type in `KibanaRequest`

restrry commented on x-pack/plugins/features/server/routes/index.test.ts:
```diff
@@ -14,7 +14,7 @@ import { RequestHandler } from '../../../../../src/core/server';
 
 describe('GET /api/features', () => {
-  let routeHandler: RequestHandler<any, any, any>;
+  let routeHandler: RequestHandler<any, any, any, any>;
```
the necessity to declare Method in generic type creates additional work for the NP plugins. what benefits did they get in return?

restrry commented on x-pack/legacy/plugins/apm/server/routes/create_api/index.ts:
```diff
@@ -65,7 +65,7 @@ export function createApi() {
         };
 
-        router[routerMethod](
+        (router[routerMethod] as RouterHandler<typeof routerMethod>)(
```
the necessity to declare Method in generic type creates additional work for the NP plugins. what benefits did they get in return?

restrry commented on x-pack/plugins/features/server/routes/index.test.ts:
```diff
@@ -14,7 +14,7 @@ import { RequestHandler } from '../../../../../src/core/server';
 
 describe('GET /api/features', () => {
-  let routeHandler: RequestHandler<any, any, any>;
+  let routeHandler: RequestHandler<any, any, any, any>;
```
the necessity to declare Method in generic type creates additional work for the NP plugins. what benefits did they get in return?

afharo replied:
Added a default type if not provided. That will make it easier to handle when we don't want to specify it :)

restrry commented on src/core/server/server.api.md:
```diff
@@ -1110,14 +1111,17 @@ export type RouteContentType = 'application/json' | 'application/*+json' | 'appl
 export type RouteMethod = 'get' | 'post' | 'put' | 'delete' | 'patch' | 'options';
 
+// @public
+export type RouterHandler<Method extends RouteMethod> = <P extends ObjectType, Q extends ObjectType, B extends ObjectType | Type<Buffer> | Type<Stream>>(route: RouteConfig<P, Q, B, Method>, handler: RequestHandler<P, Q, B, Method>) => void;
```
different type order in generic could create confusion for plugin authors. https://github.com/elastic/kibana/pull/50783/files/93d267c3910c5ccebfa10e0ec0600c7abecbe22e..7d4afae6a283634d65f6987f26ad340d946822ca?file-filters%5B%5D=.md&file-filters%5B%5D=.ts&hide-deleted-files=true#diff-f9b3398cad5ffd2f60f87bc0c65e1fedR58
we will have to create lots of `RouterHandler` and `KibanaRequest` manually during migration

afharo replied:
Moved both to the end so we maintain the order.

afharo committed:
Merge branch 'master' + fix conflicts + easier types

afharo reviewed (commented):
Fixed the conflicts with master and fixed the types so it's simpler to use now (no need to be that explicit all the time)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/12122/)
* Commit: 8efa8babcaedc58595fc9637b9fb47e8a5ef0b90

restrry commented on src/core/server/server.api.md:
```diff
@@ -746,33 +745,33 @@ export interface IUiSettingsClient {
-    // 
     // @internal
-    static from<P extends ObjectType, Q extends ObjectType, B extends ObjectType>(req: Request, routeSchemas?: RouteSchemas<P, Q, B>, withoutSecretHeaders?: boolean): KibanaRequest<P["type"], Q["type"], B["type"]>;
+    static from<P extends ObjectType, Q extends ObjectType, B extends ObjectType | Type<Buffer> | Type<Stream>>(req: Request, routeSchemas?: RouteSchemas<P, Q, B>, withoutSecretHeaders?: boolean): KibanaRequest<P["type"], Q["type"], B["type"], import("hapi").Util.HTTP_METHODS_PARTIAL_LOWERCASE>;
```
Our `core:acceptApiChanges` doesn't regenerate the docs if interfaces hasn't changed. you can enforce update with `yarn core:acceptApiChanges --docs`

afharo replied:
Fixed! Turns out, after the latest fixes, there was no need to be specific with the `typeof req.method` anymore :)

restrry commented on src/core/server/http/router/router.ts:
```diff
@@ -21,28 +21,45 @@ import { ObjectType, TypeOf, Type } from '@kbn/config-schema';
+ *
+ * @public
+ */
+export interface RouterRoute {
```
oops that was an internal interface. Now `Boom` is leaking as a valid `handler` outcome, but `handler` is an internal concept

afharo replied:
You are right! Moved it back to internal

afharo committed:
Merge branch 'master' of github.com:elastic/kibana into http-allow-payload-config-values

afharo committed:
Fix KibanaRequest.from type exposure of hapi in API docs

afharo committed:
Move RouterRoute interface back to private + Expose some public docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/12454/)
* Commit: abbba90a548081470e3e172921af1ed7819d5056

afharo committed:
Update @kbn/config-schema docs

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/12476/)
* Commit: 108fc2dc4c336a69e9bdaddc5b70a2b89df3da98

afharo commented:
Fixes #41960
